ert3/engine/_export.py

```diff
@@ -1,42 +1,94 @@
+import asyncio
 import json
+from collections import defaultdict
 from pathlib import Path
-from typing import Iterable, List, Dict, Any
-
+from typing import List, Dict, Any
 import ert
 import ert3
 
+_SOURCE_SEPARATOR = "."
 
-def _prepare_export(
-    workspace_root: Path,
+
+def _prepare_export_parameters(
+    workspace: Path,
     experiment_name: str,
-    parameter_names: Iterable[str],
-    response_names: Iterable[str],
-) -> List[Dict[str, Dict[str, ert.data.Record]]]:
-    data_mapping = [(pname, "input") for pname in parameter_names]
-    data_mapping += [(rname, "output") for rname in response_names]
+    ensemble: ert3.config.EnsembleConfig,
+    stages_config: ert3.config.StagesConfig,
+    ensemble_size: int,
+) -> Dict[str, List[ert.data.record_data]]:
+    inputs = defaultdict(list)
+    step = stages_config.step_from_key(ensemble.forward_model.stage)
+    if not step:
+        raise ValueError(f"No step for key {ensemble.forward_model.stage}")
 
-    data: List[Dict[str, Dict[str, Any]]] = []
+    for input_record in ensemble.input:
+        record_name = input_record.record
+        record_source = input_record.source.split(_SOURCE_SEPARATOR, maxsplit=1)
+        assert len(record_source) == 2
 
-    for record_name, data_type in data_mapping:
-        ensemble_record = ert.storage.get_ensemble_record(
-            workspace=workspace_root,
-            experiment_name=experiment_name,
-            record_name=record_name,
-        )
+        if record_source[0] == "storage" or record_source[0] == "stochastic":
+            exp_name = None if record_source[0] == "storage" else experiment_name
+            source = record_source[1] if record_source[0] == "storage" else None
+            collection = ert.storage.get_ensemble_record(
+                workspace=workspace,
+                record_name=record_name,
+                experiment_name=exp_name,
+                source=source,
+                ensemble_size=ensemble_size,
+            )
+            # DO NOT export blob records as inputs
+            if collection.record_type == ert.data.RecordType.BYTES:
+                continue
+            for record in collection.records:
+                inputs[record_name].append(record.data)
 
-        if not data:
-            data = [{"input": {}, "output": {}} for _ in ensemble_record.records]
+        elif record_source[0] == "resources":
+            record_mime = next(
+                input_.mime for input_ in step.input if input_.record == record_name
+            )
+            # DO NOT export blob records as inputs
+            if record_mime == "application/octet-stream":
+                continue
+            file_path = workspace / "resources" / record_source[1]
```

Comment:
Use `os.path.join` ...

Reply:
Why? I thought the recommended way was to use pathlib.Path("some_path") / "something_else"?

Reply:
Right, missed that `workspace` is a `Path` :) It's alright then...